src/main/java/it/gov/pagopa/payhub/activities/service/exportflow/debtposition/BaseExportFlowFileService.java

```diff
@@ -0,0 +1,206 @@
+        this.exportFileErrorArchiverService = exportFileErrorArchiverService;
+        this.fileArchiverService = fileArchiverService;
+        this.workingDirectory = workingDirectory;
+        this.relativeFileFolder = relativeFileFolder;
+        this.fileNamePrefix = fileNamePrefix;
+    }
+
+    public ExportFlowFileResult executeExport(Long exportFileId){
+        int[] pageNumber = {0};
+        long exportedRows = 0;
+        long errorCount = 0;
+        List<ExportFileErrorDTO> errorMessages;
+
+        E exportFileRecord = findExportFileRecord(exportFileId);
+        Long organizationId = getOrganizationId(exportFileRecord);
+        String profile = getFlowFileVersion(exportFileRecord);
+        Path csvFilePath = workingDirectory.resolve(String.valueOf(organizationId)).resolve(relativeFileFolder).resolve(getExportFileName(exportFileId));
+
+        if (checkStatusExportFileRecord(exportFileRecord)){
+            try {
+                F exportFilter = getExportFilter(exportFileRecord);
+                List<C> csvRows = retrieveAndMapPage(exportFileRecord, exportFilter, pageNumber[0]++);
+                if (!csvRows.isEmpty()){
+                    exportedRows += csvRows.size();
+                    csvService.createCsv(csvFilePath, csvRowDtoClass, () -> csvRows, profile);
+                }
+
+            } catch (IOException e) {
+                errorCount ++;
+                errorMessages = new ArrayList<>();
+                errorMessages.add(ExportFileErrorDTO.builder()
+                        .fileName(getExportFileName(exportFileId))
+                        .errorCode("Export Exception")
+                        .errorMessage(e.getMessage())
+                        .build());
+
+                log.error("Error processing creation csv file with exportFileId {}: {}", exportFileId, e.getMessage());
+                archiveErrorFiles(workingDirectory, exportFileId, organizationId, errorMessages);
+            }
+        }
+
+        ExportFlowFileResult exportFlowFileResult = ExportFlowFileResult.builder()
+                .fileName(getExportFileName(exportFileId))
+                .exportedRows(exportedRows - errorCount)
+                .filePath(csvFilePath.toString())
+                .build();
+
+        createZipArchive(exportFlowFileResult);
+
+        return exportFlowFileResult;
+    }
+
+    private void createZipArchive(ExportFlowFileResult exportFlowFileResult) {
+        try {
+            Path tmpZipFilePath = Path.of(exportFlowFileResult.getFilePath())
+                    .getParent()
+                    .resolve(Utilities.replaceFileExtension(exportFlowFileResult.getFileName(), ".zip"));
+            fileArchiverService.compressAndArchive(List.of(Path.of(exportFlowFileResult.getFilePath())), tmpZipFilePath, workingDirectory);
+        } catch (IOException e) {
+            throw new InvalidExportFileException("Error during compression and archiving: " + e.getMessage());
```

Comment:
This is not an invalid file error... if this happen, something related to the container permission on the file system is wrong...
This should be a retryable exception, let's handle it with a IllegalStateException (and keep the original exception as throwable related to the new exception)